### PR TITLE
Geometry Engine tests and cleanup

### PR DESCRIPTION
**Pull request by pgrinaway**
This PR will 

* Complete the `GeometryEngine` tests
* Fix any test failures
* Return to sampling torsions from a pdf

Right now, the openeye problem seems to have reared its head again.

### CODE REVIEW

**Comment by pgrinaway** [comment]
Ok, going to go through and see if I can fix the openeye thing:

* Confirmed I am running Python 3.4

* Confirmed the `OEMol` has no bonds when it is given to `OEAssignPartialCharges`

* `generateTopologyFromOEMol` seems to produce a topology with all bonds from one atom to itself. Will move discussion to openmoltools.

**Comment by pgrinaway** [comment]
Actually, I think I see the problem.

**Comment by pgrinaway** [comment]
I neglected to 
```python
    oechem.OETriposAtomNames(mol)
    oechem.OETriposBondTypeNames(mol)
```

**Comment by pgrinaway** [comment]
(debugging unit madness at the moment)

**Comment by pgrinaway** [comment]
Found an error... will fix now.

File: perses/rjmc/geometry.py [hunk 1]
```diff
@@ -163,30 +164,31 @@ def logp_reverse(self, top_proposal, new_coordinates, old_coordinates):
...
-            reverse_proposal_coordinates[atom.idx] = new_coordinates[new_index]
+        old_structure = parmed.openmm.load_topology(top_proposal.old_topology, top_proposal.old_system)
+        old_atoms_with_positions = [old_structure.atoms[atom_idx] for atom_idx in top_proposal.old_to_new_atom_map.keys()]
+        old_unique_atoms = [old_structure.atoms[idx] for idx in top_proposal.unique_old_atoms]
+
+        #copy common atomic positions
+        for atom in old_structure.atoms:
+            if atom.idx in old_atoms_with_positions:
+                corresponding_new_index = top_proposal.old_to_new_atom_map[atom.idx]
+                old_coordinates[atom.idx] = new_coordinates[corresponding_new_index]
```
Comment (jchodera) [inline perses/rjmc/geometry.py]: This looks fishy to me.  What are you trying to do here---shouldn't you already have the old coordinates?

The goal of computing the reverse proposal probability is just to compute the probability of placing the old atoms that are deleted back where they were originally.

Comment (jchodera) [inline perses/rjmc/geometry.py]: Ah---I think I understand.  You're trying to reuse as much code as possible to go through the same sequence of placing the old atoms one at a time?

**Comment by pgrinaway** [comment]
definitely some bond length weirdness in the reverse logp..

File: perses/rjmc/geometry.py (hunk 1 above)
Comment (pgrinaway) [inline perses/rjmc/geometry.py]: Yeah, I was trying to reduce code duplication. I realize I have a bit of a ways to go on that front, but I was hoping it'd help solve some of the weird problems.

File: perses/rjmc/geometry.py [hunk 2]
```diff
@@ -198,31 +200,34 @@ def logp_reverse(self, top_proposal, new_coordinates, old_coordinates):
...
                 angle_k = angle.type.k
                 sigma_theta = units.sqrt(1/(beta*angle_k))
                 logZ_theta = np.log((np.sqrt(2*np.pi)*sigma_theta/sigma_theta.unit)) #need to eliminate unit to allow numpy log
-                logp_theta = self._angle_logq(internal_coordinates[1], angle, beta) - logZ_theta
+                logp_theta = self._angle_logq(theta, angle, beta) - logZ_theta
 
                 #calculate torsion probability
-                logp_phi = self._torsion_logp(atom, atom_coords, torsion, atoms_with_positions, reverse_proposal_coordinates, beta)
+                logp_phi = self._torsion_logp(atom, atom_coords, torsion, old_atoms_with_positions, old_coordinates, beta)
                 logp = logp + logp_choice + logp_r + logp_theta + logp_phi + np.log(detJ)
```
Comment (jchodera) [inline perses/rjmc/geometry.py]: Do you know which of these terms is responsible for the huge magnitude of the reverse `logp`?

Comment (pgrinaway) [inline perses/rjmc/geometry.py]: yeah, it was the bond. The length was cut by a factor of 10 as a result of a unit error.

**Comment by pgrinaway** [comment]
Ok, now the `logp_bond` is only ~2, but `logp_phi` is ~-75 for the reverse when it's more like -4 for the forward proposal. Investigating...

**Comment by pgrinaway** [comment]
I think it might be another unit error. One sec.

**Comment by pgrinaway** [comment]
I'm going to merge this and then make sure the `GeometryEngine` API is consistent with the rest of the package (fixing the build failures) then add the last tests.

**Pull request merged**